### PR TITLE
Mon core facade impl

### PR DESCRIPTION
**Pull request by acalabro**
Rest Notification to the CorePlatform through Post methods defined in 

learnpad/lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/SimulatorMonCoreFacadeRESTResource.java

Parameters are defined in learnpad/lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/utils/RestResource

Those classes will be used by monitoring through the eu.learnpad.simulator.mon.manager.RestNotifier module enacted by ComplexEventProcessor once events from simulator will flow in and a rule will be triggered.

[Review on Reviewable](https://reviewable.io/reviews/learnpad/learnpad/247)

### CODE REVIEW

File: lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/SimulatorMonCoreFacadeRESTResource.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+		//params.add(new BasicNameValuePair("param-2", "n1hehe"));
+		try {
+		    PostMethod method = new PostMethod(uri);
+			response = httpClient.executeMethod(method);
+			
+		} catch (IOException e) {
+			DebugMessages.println(TimeStamp.getCurrentTime(), 
+					SimulatorMonCoreFacadeRESTResource.class.getCanonicalName(),
+					"Exception in receiveSimulationStartEvent\nResponse: " + response);
+			e.printStackTrace();
```
Comment (gulyx) [inline lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/SimulatorMonCoreFacadeRESTResource.java]: @acalabro i suggest to do not print stack trace here. the operation is supposed to launch an exception if something wrong happens, so let's implement it accordingly .... If you are fine with this comment, please fix all the instances

in cc : @tomjorquera, @woshilapin

File: lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/utils/RestResource.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.apache.commons.httpclient.auth.AuthScope;
+
+public class RestResource {
+	final private static String PROTOCOL = "http";
+	final private static String HOSTNAME = "localhost";
+	final private static int PORT = 8080;
+	final private static int SIM_PORT = 8081;
+	final private static String HOST = PROTOCOL + "://" + HOSTNAME + ":" + PORT;
+	final private static String SIM_HOST = PROTOCOL + "://" + HOSTNAME + ":" + SIM_PORT;
+	final private static String DEFAULT_USER = "userName";
```
Comment (woshilapin) [inline lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/utils/RestResource.java]: You should use `superadmin` as a user.

File: lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/utils/RestResource.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	final private static String DEFAULT_PASSWORD = "LearnPAss";
+	final public static String REST_URI = HOST + "/sim/mon/rest";
```
Comment (woshilapin) [inline lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/utils/RestResource.java]: If this is the Core Platform you want to contact, the prefix is `/xwiki/rest/` and you may have to append `/learnpad/sim/corefacade` for all the REST services for the Controller of Simulation Environment.

File: lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/SimulatorMonCoreFacadeRESTResource.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	@Override
+	public UserData getUserData(String userartifactid) {
+		return null;
+	}
+
+	@Override
+	public void receiveSimulationStartEvent(SimulationStartEvent event) throws LpRestException {
+		HttpClient httpClient = RestResource.getClient();
+		String uri = String.format("%s/simulationstart/%s",
+				RestResource.REST_URI, event);
```
Comment (woshilapin) [inline lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/SimulatorMonCoreFacadeRESTResource.java]: Event should not be part of the URL.  It's a Java class that will be transformed into JSON.  It should be part of the body so you have to transform the Java object to JSON. And after, you'll have to create a HttpEntity that you will set in the Post method with `.setEntity()`.

File: lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/SimulatorMonCoreFacadeRESTResource.java [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+	@Override
+	public UserData getUserData(String userartifactid) {
+		return null;
+	}
+
+	@Override
+	public void receiveSimulationStartEvent(SimulationStartEvent event) throws LpRestException {
+		HttpClient httpClient = RestResource.getClient();
+		String uri = String.format("%s/simulationstart/%s",
```
Comment (woshilapin) [inline lp-simulation-environment/monitoring/src/main/java/eu/learnpad/simulator/api/impl/SimulatorMonCoreFacadeRESTResource.java]: All the URL should be revised in `lp-cp-apis` (look [here](https://github.com/LearnPAd/learnpad/blob/master/lp-core-platform/lp-cp-apis/src/main/java/eu/learnpad/sim/rest/IEventReceiver.java)).

**Comment by gulyx** [comment]
@woshilapin, it seems that the last commit from @acalabro failed due to the "usual issue" with the CA, right? should he add a dummy commit in order to trigger again travis?

**Comment by woshilapin** [comment]
Do you still have commits to do on this branch or can I merge it if Travis success the build?

**Comment by acalabro** [comment]
@woshilapin no more commits, if for you is ok, we can merge it.
ty.

**Comment by woshilapin** [comment]
Cool, thank you :smile:

**Pull request merged**